docs/docs/developers/development/dialogs/index.md

```diff
@@ -64,4 +69,103 @@ It will result with following in UI dialog:
+Front End validation is done by Atlaskit by default, if the submitted value is obviously incorrect, like a non-number value in NumberField. You can find more details [here](https://atlassian.design/components).
+![](dialog-frontend-validation.png)
+
+### BackEnd validation
+WebSight supports the validation of dialog values on BackEnd side. If the value is incorrect, then it won’t be saved and the dialog can’t be submitted. The author will see a proper error message.
+![](dialog-backend-validation.png)
+
+## Show/hide dialog fields
+By default, all dialog components are visible, but there is a possibility to hide them.
+
+### Context
+To show or hide a particular field depending on dialog context you can use a `ws:disallowedContext` parameter.
+
+```json
+"ws:disallowedContext": ["edit"]
+```
+
+To hide an element in dialog, the request from Front-End which fetches it has to contain the additional parameter `context`. If the context value matches one of `ws:dissallowedContext` values, then the field won’t be rendered.
+
+### Conditions
+To show or hide a particular field depends on other fields’ state you can use a `ws:display` node.
+This node should contain children defining conditions to show the element. If the component has such a child node it’s hidden by default. It’s required to fulfill at least one condition to show the component.
+
+Each condition should have two properties:
+- sourceName - with the name of the component whose value would be checked
+- values - with one or more values. At least one of them should match the source field value to fulfill the condition.
+
+Example conditions configurations:
+- with single value:
+```json
+"ws:display": {
+  "condition": {
+    "sourceName": "fieldName",
+    "values": "option1"
+  }
+}
+```
+- with multiple values:
+```json
+"ws:display": {
+  "condition": {
+    "sourceName": "fieldName",
+    "values": ["option1", "option2", "option3"]
+  }
+}
+```
+
+Example dialog definition:
+
+```json
+{
+  "sling:resourceType": "wcm/dialogs/dialog",
+  "hideall": {
+    "sling:resourceType": "wcm/dialogs/components/toggle",
+    "name": "hideall",
+    "label": "Hide all other elements"
+  },
+  "container": {
+    "sling:resourceType": "wcm/dialogs/components/container",
+    "showrequiredfield": {
+      "sling:resourceType": "wcm/dialogs/components/toggle",
+      "name": "showrequiredfield",
+      "label": "Show required field"
+    },
+    "requiredfield": {
+      "sling:resourceType": "wcm/dialogs/components/textfield",
+      "name": "requiredfield",
+      "label": "Required field",
+      "required": true,
+      "ws:display": {
+        "condition": {
+          "sourceName": "showrequiredfield",
+          "values": "true"
+        }
+      }
+    },
+    "ws:display": {
+      "condition": {
+        "sourceName": "hideall",
+        "values": "false"
+      }
+    }
+  }
+}
+```
+![](dialog-show-hide-1.png)
+![](dialog-show-hide-2.png)
+![](dialog-show-hide-3.png)
```

Comment:
let's add some comment on the images and describe what is presented/what case is visible